build/commands/scripts/generateCoverageReport.js

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+const { glob, rm } = require('fs/promises')
+const { writeJSON, mkdirp } = require('fs-extra')
+const utils = require('../lib/util')
+const config = require('../lib/config')
+
+module.exports = (program) =>
+  program
+    .command('coverage_report')
+    .description(
+      [
+        'generates a coverage report.',
+        'Requires to build with --is_coverage and run test.',
+        'HTML report will be in out/$BUILD_DIR/coverage/report/$NAME/dist',
+      ].join('\n'),
+    )
+    .option('-C [build_dir]', 'build config (out/Debug, out/Release)')
+    .option('--build_type [build_type]', 'build with coverage information ')
+    .option('--target_arch [target_arch]', 'target architecture')
+    .option('--target_os <target_os>', 'target OS')
+    .option('--name [name]', 'name of the test report [defaults to coverage]', 'all')
+    .option(
+      '--clean',
+      'delete recordings and reports. Either all or those that match --tests [testSuites]',
+    )
+    .option(
+      '--tests [testSuites]',
+      'comma seperated list of testsuites to consider. By default it parses all recordings',
+    )
+    .arguments('[build_config]')
+    .action(async (buildConfig, args) => {
+      config.buildConfig = buildConfig || config.buildConfig
+      config.update(args)
+      const dirsGlob = args.tests ? `{${args.tests}}/**` : '**'
+
+      const out = `${config.outputDir}/coverage`
+      const reportPath = `${out}/report/${args.name}`
+      const profDataPath = `${reportPath}/coverage.profdata`
+      const distPath = `${reportPath}/dist`
+
+
+      const recordings = await Array.fromAsync(
+        glob(`${out}/${dirsGlob}/*.profraw`),
+      )
+
+      if (args.clean) {
+        await Promise.all(
+          recordings
+            .concat([reportPath])
+            .map((path) => rm(path, { recursive: true, force: true })),
+        )
+
+        return
+      }
+
+      const allTestSuites = [
+        ...new Set(
+          recordings
```

Comment:
please add some comments so I don't have to figure out what this is doing :)

Reply:
sry. I didn't see that you might be referring to the whole transformation chain. will factor out to a more descriptive functions